src/lru.ml

```diff
@@ -36,25 +48,31 @@ struct
   let find_opt t k =
     try
       let result = find t k in
+      Stats.hit t.stats;
       Some result
-    with Not_found -> None
+    with Not_found ->
+      Stats.miss t.stats;
+      None
 
   let promote t k = ignore (find t k)
 
   let replace t k v =
     try
       let index, _value = H.find t.tbl k in
       let new_index = Dllist.promote t.lst index in
-      H.replace t.tbl k (new_index, v)
+      H.replace t.tbl k (new_index, v);
+      Stats.replace t.stats
     with Not_found ->
       let index, removed = Dllist.append t.lst k in
       (match removed with None -> () | Some key -> H.remove t.tbl key);
-      H.replace t.tbl k (index, v)
+      H.replace t.tbl k (index, v);
+      Stats.add t.stats
```

Comment:
Instead of `Stats.replace` and `Stats.add`, what do you think about the following?
- `replace_existing`, for situations where `k` is in `t`
- `replace_add` for situations where `k` is not in `t` and `t` is not full
- `replace_evict` for situations where `k` is not in `t` and `t` is full

Reply:
I see thanks. 

Having `replace_existing` had the advantage of making possible the count of calls to `replace`. If this is removed, we can't count the total number of calls to `replace`. WDYT @pascutto

Reply:
I think I will add again a replace counter in order to have `add`, `discard` and `replace`.